pkg/webhook/webhook.go

```diff
@@ -546,6 +547,68 @@ func createNodeSelectorPatch(patch []jsonPatchOperation, existing map[string]str
 	return patch
 }
 
+func createCustomizedPatch(pod corev1.Pod) ([]jsonPatchOperation, error) {
+	var patch jsonPatchOperation
+	var customizedPatch []jsonPatchOperation
+
+	for cik, civ := range customizedInjections {
+		// The customizedInjections will be injected when:
+		// 1. Pod labels contain the patch key defined in customizedInjections, and
+		// 2. The value of patch key in pod labels(not in customizedInjections) is "true"
+		if podValue, exists := pod.ObjectMeta.Labels[cik]; exists && podValue == "true" {
+			err := json.Unmarshal([]byte(civ), &patch)
+			if err != nil {
+				return nil, err
+			}
+		}
+		customizedPatch = append(customizedPatch, patch)
+	}
+	for _, p := range customizedPatch {
+		// metadata.Annotations is the only supported field for customization
+		// jsonPatchOperation.Path should be "/metadata/annotations" when customizing pod annotation
+		if p.Path != "/metadata/annotations" {
+			err := errors.New("path:" + p.Path + " is not supported, only /metadata/annotations can be customized")
+			return nil, err
+		}
+	}
+	return customizedPatch, nil
+}
+
+func getNetworkSelections(netType string, pod corev1.Pod, customizedPatch []jsonPatchOperation) (string, bool) {
+	var annotateKey string
+	switch netType {
+	case "default":
+		annotateKey = defaultNetworkAnnotationKey
+	case "additional":
+		annotateKey = networksAnnotationKey
+	}
+
+	// User defined annotateKey takes precedence than customized injections
+	glog.Infof("search %s net-attach-def in original pod annotations", netType)
```

Comment:
Nit: can't we just log annotateKey instead of netType ? this way networkAnnotationKey can be passed from calling side (instead of netType) which avoids above switch case.

Reply:
Yes, good catch! updated.